extensions/amp-analytics/0.1/test/test-vendors.js

```diff
@@ -124,27 +127,17 @@ describes.realWin('amp-analytics', {
             sandbox.stub(urlReplacements.getVariableSource(), 'get').callsFake(
                 function(name) {
                   expect(this.replacements_).to.have.property(name);
-
                   const defaultValue = `_${name.toLowerCase()}_`;
-                  const extraMapping = VENDOR_REQUESTS[vendor][name];
                   return {
-                    sync: paramName => {
-                      if (!extraMapping ||
-                          extraMapping[paramName] === undefined) {
-                        return defaultValue;
-                      }
-                      return extraMapping[paramName];
-                    },
+                    sync: () => defaultValue,
                   };
                 });
 
-            const variables = variableServiceFor(ampdoc.win);
-            const {encodeVars} = variables;
-            sandbox.stub(variables, 'encodeVars').callsFake(
-                function(name, val) {
-                  val = encodeVars.call(this, name, val);
-                  if (val == '') {
-                    return '$' + name;
+            sandbox.stub(ExpansionOptions.prototype, 'getVar').callsFake(
+                function(name) {
+                  let val = this.vars[name];
+                  if (val == null || val == '') {
+                    val = '!' + name;
```

Comment:
emmm I feel `!` is not so intuitive as `$`. We may want to document how to generate the test results to vendors.

Reply:
right, but `$` will be urlencoded. I found `!` is not used in any vendor config, so it's good.